runtime/onert/core/include/backend/basic/BackendContextHelpers.h

```diff
@@ -219,25 +243,44 @@ template <typename T_BackendContext> ITensorRegistry *genTensors(T_BackendContex
 inline void initConsts(const ir::Operands &operands,
                        const util::Set<ir::OperandIndex> &external_operands,
                        ITensorRegistry *tensor_registry,
-                       const ir::OperandIndexMap<ir::OperandIndex> & /*shared_memory_operands_map*/)
+                       const ir::OperandIndexMap<ir::OperandIndex> &shared_memory_operands_map)
 {
   operands.iterate([&](const ir::OperandIndex &ind, const ir::Operand &operand) {
-    if (external_operands.contains(ind) || !operand.isConstant())
+    const bool has_const_shared_memory =
+      shared_memory_operands_map.find(ind) != std::end(shared_memory_operands_map) &&
+      operands.at(shared_memory_operands_map.at(ind)).isConstant();
+    const bool can_be_initialized_as_const = operand.isConstant() || has_const_shared_memory;
+    if (external_operands.contains(ind) || !can_be_initialized_as_const)
       return;
 
     auto tensor = tensor_registry->getNativeITensor(ind);
     assert(tensor != nullptr);
 
     VERBOSE(FillOperandData) << "Fill data for " << ind << std::endl;
 
-    auto data = operand.shareData();
-    assert(data && data->base());
-    ExternalTensor *ext_tensor = dynamic_cast<ExternalTensor *>(tensor);
-
-    if (ext_tensor == nullptr)
-      throw std::runtime_error{"This tensor is not external tensor"};
-
-    ext_tensor->setData(data);
+    if (has_const_shared_memory)
+    {
+      const auto &source_operand_ind = operands.at(shared_memory_operands_map.at(ind));
+      auto memory_source_data = source_operand_ind.shareData();
+      assert(memory_source_data && memory_source_data->base());
+      auto shared_mem_tensor = dynamic_cast<Tensor *>(tensor);
```

Comment:
All tensors generated by cpu backend cannot be `nullptr`. In other words, All tensors inherited from `basic::Tensor` are not `nullptr`. Is this casting meaningful?

Reply:
Note that `tensor` here has `ITensor` type and needed `setBuffer` method is not exposes in `ITensor` interface.

> All tensors inherited from basic::Tensor are not nullptr

Agree that just an `assert` should be enough here

Reply:
Good tip but it requires changing `ITensorRegistry *tensor_registry` argument of `initConsts` to `TensorRegistry`. Since `initConsts` is called in many places, such changes requires a pretty big rector (it should be probably implemented as a separate PR).

Reply:
@ragmani I've tried to resolve https://github.com/Samsung/ONE/blob/913cfa92ec9705675491d65390237383f62075a6/runtime/onert/backend/cpu/BackendContext.cc#L50 but especially due to public field [tensor_registry](https://github.com/Samsung/ONE/blob/913cfa92ec9705675491d65390237383f62075a6/runtime/onert/core/include/backend/BackendContext.h#L75) it requires a lot of changes in all backends ;(

Reply:
OK, let's consider it later.